src/openloco/map/tilemgr.cpp

```diff
@@ -211,6 +221,17 @@ namespace openloco::map::tilemgr
     // 0x0046112C
     void map_invalidate_map_selection_tiles()
     {
-        call(0x0046112C);
+        // 2=MAP_SELECT_FLAG_ENABLE_CONSTRUCT - from RCT2 code
+        if ((input::getMapSelectionFlags() & 2) == 0)
+            return;
+
+        for (uint16_t index = 0;; ++index)
+        {
+            auto& position = _mapSelectedTiles[index];
+            if (position.x == -1)
+                break;
+            assert(index >= 0 && index < mapSelectedTilesSize);
```

Comment:
bit late for this when you have already dereferenced. I would change the loop so that it is `for(auto i = 0; i < mapSelectedTilesSize; ++i)`. I don't really see there being much need to have an assert for this.

Reply:
I think it's useful in case I made a mistake with decoding in size of the array and the array is bigger in the original code. What do you think?

Reply:
Ok, updated in the latest commit.